index.js

```diff
@@ -0,0 +1,91 @@
+import express from 'express';
+import fs from 'fs';
+
+const server = express();
+
+let requestExpirationDates = [];
+let nextExpirationDate = new Date();
+let nextComparsionInterval = 1;
+
+setInterval(() => { 
+    if(nextExpirationDate <= new Date() && requestExpirationDates.length !== 0){
+      requestExpirationDates.every(element => {
+        if(element <= nextExpirationDate){
+          requestExpirationDates.shift();
+          updateNextExpirationDateAndComparsionInterval();
+        }
+        else
+          return false;
+      });
+      writeOnFile();
+    }
+
+}, nextComparsionInterval);
+
+
+readFileAndUpdateNumberOfRequests();
+
+server.get('/requestCounter', (req, res) =>{
+
+  let expirationDate = new Date(new Date().getTime() + 1000 * 60);
+  if(requestExpirationDates.length === 0){
+    nextExpirationDate = expirationDate;
+  }
+  requestExpirationDates.push(expirationDate);
+
+  getRequests().then(
+    (val) => { 
+      return res.json(val);
+    }
+  )
+});
+
+function getRequests() {
+  return new Promise((resolve) => {
+    writeOnFile();
+    resolve(requestExpirationDates.length);
+  })
+}
+
+function writeOnFile() {
+  fs.writeFile('./assets/requests.txt', requestExpirationDates.toString(), (err) => {
```

Comment:
Storing request dates in a file is a good solution to the problem of counting across server restarts :+1: